controllers/k8ssandra/cleanup.go

```diff
@@ -153,9 +153,144 @@ func (r *K8ssandraClusterReconciler) checkDcDeletion(ctx context.Context, kc *ap
 }
 
 func (r *K8ssandraClusterReconciler) deleteDc(ctx context.Context, kc *api.K8ssandraCluster, dcName string, logger logr.Logger) result.ReconcileResult {
+
+	// delete Stargate
+	if result := r.deleteStargate(ctx, kc, dcName, logger); result != nil {
+		return result
+	}
+
+	// delete Reaper
+	if result := r.deleteReaper(ctx, kc, dcName, logger); result != nil {
+		return result
+	}
+
+	// delete the DC
+	if result := r.deleteDatacenter(ctx, kc, dcName, logger); result != nil {
```

Comment:
We need to improve naming here, there's a `deleteDc` and a `deleteDatacenter` and they do different things.
Maybe `deleteDcComponents` for the first one, and `deleteCassandraDatacenter` for the second (matches the name of the CR that it deletes)?